astrolabe/utils.py

```diff
@@ -261,14 +264,14 @@ def read_stats(self):
             LOGGER.info("Reading sentinel file {!r}".format(self.sentinel))
             with open(self.sentinel, 'r') as fp:
                 stats = json.load(fp)
+                LOGGER.info("Sentinel contains: %s" % json.dumps(stats))
+                return stats
         except FileNotFoundError:
             LOGGER.error("Sentinel file not found")
-            stats = self._PLACEHOLDER_EXECUTION_STATISTICS
+            raise WorkloadExecutorError("The workload executor did not write a results.json file in the expected location")
```

Comment:
The workload executor is always expected to produce a (presumably valid) `results.json` file, so I removed `_PLACEHOLDER_EXECUTION_STATISTICS` in favor of an explicit error. This also guarantees to the caller that any returned stats will have originated from a real file produced by the workload executor.